src/main/java/uk/gov/justice/hmpps/prison/service/OffenderAssessmentService.java

```diff
@@ -100,9 +102,20 @@ private AssessmentSummary getAssessmentSummary(final OffenderAssessment assessme
         }
 
         final var assessmentItems = assessmentDetails.getAssessmentItems();
-        final var assessmentAnswersByQuestionId = assessmentItems.stream().map(OffenderAssessmentItem::getAssessmentAnswer).collect(Collectors.toMap((aa) -> aa.getParentAssessment().getAssessmentId(), AssessmentEntry::getDescription));
+        final var assessmentAnswersByQuestionId = assessmentItems.stream()
+            .map(OffenderAssessmentItem::getAssessmentAnswer)
+            .collect(groupingBy((aa) -> aa.getParentAssessment().getAssessmentId(), mapping(AssessmentEntry::getDescription, toList())));
+
+        return assessmentQuestions.stream().map(aq -> getAssessmentQuestionAndAnswers(aq, assessmentAnswersByQuestionId.get(aq.getAssessmentId()))).collect(toList());
+    }
 
-        return assessmentQuestions.stream().map(aq -> new AssessmentQuestion(aq.getDescription(), assessmentAnswersByQuestionId.get(aq.getAssessmentId()))).collect(Collectors.toList());
+    private AssessmentQuestion getAssessmentQuestionAndAnswers(final AssessmentEntry assessment, final List<String> answers) {
+        if (answers == null) {
+            return new AssessmentQuestion(assessment.getDescription(), null, null);
```

Comment:
is it worth having a test covering when there aren't any answers?

Reply:
There already is a test for that, as part of the main test. See https://github.com/ministryofjustice/prison-api/blob/main/src/test/java/uk/gov/justice/hmpps/prison/service/OffenderAssessmentServiceTest.java#L132